### PR TITLE
Add support for a working att/ast project -- AT&T kornshell -- for the solaris 11.4 platform

### PR DESCRIPTION
**Pull request by sneyx1234**

### CODE REVIEW

File: mesonbuild/compilers/compilers.py [hunk 1]
```diff
@@ -1291,6 +1291,11 @@ def build_unix_rpath_args(self, build_dir, from_dir, rpath_paths, build_rpath, i
                     paths = paths + ':' + padding
             args.append('-Wl,-rpath,' + paths)
 
+	# BBI_SOL11_4 BEGIN -- just to prevent code of get_compiler_is_linuxlike below -- linux is like sunos and not vice versa
```
Comment (ignatenkobrain) [inline mesonbuild/compilers/compilers.py]: __[Flake8]__ [W191] indentation contains tabs ([link](https://sider.review/gh/repos/19784232/pulls/5350#issue-4029585))

<sub>posted by Sider</sub>

File: mesonbuild/compilers/compilers.py [hunk 2]
```diff
@@ -1291,6 +1291,11 @@ def build_unix_rpath_args(self, build_dir, from_dir, rpath_paths, build_rpath, i
...
+        if mesonlib.is_sunos():
+            return args
+	# BBI_SOL11_4 END
```
Comment (ignatenkobrain) [inline mesonbuild/compilers/compilers.py]: __[Flake8]__ [W191] indentation contains tabs ([link](https://sider.review/gh/repos/19784232/pulls/5350#issue-4029588))

<sub>posted by Sider</sub>

File: mesonbuild/compilers/compilers.py [hunk 3]
```diff
@@ -1291,6 +1291,11 @@ def build_unix_rpath_args(self, build_dir, from_dir, rpath_paths, build_rpath, i
                     paths = paths + ':' + padding
             args.append('-Wl,-rpath,' + paths)
 
+	# BBI_SOL11_4 BEGIN -- just to prevent code of get_compiler_is_linuxlike below -- linux is like sunos and not vice versa
+        if mesonlib.is_sunos():
```
Comment (ignatenkobrain) [inline mesonbuild/compilers/compilers.py]: __[Flake8]__ [E101] indentation contains mixed spaces and tabs ([link](https://sider.review/gh/repos/19784232/pulls/5350#issue-4029586))

<sub>posted by Sider</sub>

File: mesonbuild/compilers/compilers.py (hunk 2 above)
Comment (ignatenkobrain) [inline mesonbuild/compilers/compilers.py]: __[Flake8]__ [E101] indentation contains mixed spaces and tabs ([link](https://sider.review/gh/repos/19784232/pulls/5350#issue-4029587))

<sub>posted by Sider</sub>

File: mesonbuild/compilers/compilers.py (hunk 1 above)
Comment (ignatenkobrain) [inline mesonbuild/compilers/compilers.py]: __[Flake8]__ [E101] indentation contains mixed spaces and tabs ([link](https://sider.review/gh/repos/19784232/pulls/5350#issue-4029584))

<sub>posted by Sider</sub>

Comment (sneyx1234) [reply mesonbuild/compilers/compilers.py]: Tabs cleaned with commit 21c3e12

**Comment by sneyx1234** [comment]
The merge into the master is required for seamless integration of the att/ast project.

File: mesonbuild/mesonlib.py [hunk 4]
```diff
@@ -346,6 +346,11 @@ def __setitem__(self, machine: MachineChoice, val: typing.Optional[_T]) -> None:
         }[machine]
         setattr(self, key, val)
 
+# BBI_SOL11_4 BEGIN -- added definition for platform specific conditions
```
Comment (dcbaker) [inline mesonbuild/mesonlib.py]: we don't use comments like this in meson.

File: mesonbuild/mesonlib.py [hunk 5]
```diff
@@ -346,6 +346,11 @@ def __setitem__(self, machine: MachineChoice, val: typing.Optional[_T]) -> None:
...
+def is_sunos() -> bool:
+    return platform.system().lower() == 'sunos'
```
Comment (dcbaker) [inline mesonbuild/mesonlib.py]: there was some debate a while ago whether we should call this `sunos` or `solaris`, @jpakkane, I'm okay with sunos at this point and I'm willing to just make the call, does that work for you?

File: mesonbuild/compilers/compilers.py (hunk 3 above)
Comment (dcbaker) [reply mesonbuild/compilers/compilers.py]: looks like this line may still have tabs in it?

**Comment by dcbaker** [comment]
I'm happy then as long as @jpakkane is.

**Comment by jpakkane** [comment]
The fd closing still remains. As said earlier it was changed because the old behaviour (which this commit brings back) was broken for people. Why is the change needed?

And for the sunos/solaris naming: this change is internal to Meson. We can easily change it later. The discussion was about the name we expose inside Meson files with, e.g. `host_machine.system()`.

**Comment by jpakkane** [comment]
:crossed_swords: conflicts. This is probably because you have merged from master to your branch. Please don't do that, keep the MR up to date by rebasing instead. Merges lead to messy history and merge problems such as these.

**Comment by ignatenkobrain** [comment]
Flake8 detected 26 issues on 71079701783a782496eb681c42a212394546032d
Visit https://sider.review/gh/repos/19784232/pulls/5350 to review the issues.

<sub>posted by Sider</sub>

**Pull request closed without merging**